internal/handlers/common/error.go

```diff
@@ -246,13 +246,44 @@ func (we *WriteErrors) Document() *types.Document {
 	return d
 }
 
+// parseArr converts err to a T type.
+// If it fails, the returned bool value is false.
+func parseErr[T any](err error) (T, bool) {
+	var obj T
+	ok := errors.As(err, &obj)
+
+	return obj, ok
+}
+
+// Append convert the err to the writeError type and
+// appends it to WriteErrors.
+func (we *WriteErrors) Append(err error) {
+	if e, ok := parseErr[*writeError](err); ok {
+		*we = append(*we, *e)
+		return
+	}
+
+	if e, ok := parseErr[*CommandError](err); ok {
+		*we = append(*we, writeError{err: e.Unwrap().Error(), code: e.code})
+		return
+	}
+
+	*we = append(*we, writeError{err: err.Error(), code: errInternalError})
```

Comment:
I don't understand why it's exactly `errInternalError`. I guess, in the general case, it could be any type of error code?

Reply:
The point is that errors that are not `writeError` or `commandError` don't have any error code attached. When I was talking with @w84thesun we agreed that internalError represents these errors the best. Of course we can try to find another solution.